### ISSUE TITLE
How modify apache configuration?

### ISSUE THREAD

**Issue by iferca** [nextcloud/nextcloud-snap#1651]
How can I enable apache text compression and http2 in a read-only snap installation of nextcloud. 
Website is crazy slow and I noticed that when analyzing performance, transporting uncompressed data is the largest time followed but the recommendation to enable http2

[Could not fin any `question` type of issue, please let me know if there is another channel to ask questions like this]

**Comment by kyrofa** [comment]
The short answer is: you can't.

The long answer is that we keep those things read-only for a few reasons:

1. Practicality: snaps bundle their dependencies, and users can't just put more stuff inside them. As a result, it's entirely possible that users can select a configuration that has a dependency the snap can't satisfy, or perhaps the snap accidentally satisfies and that dependency is removed later, thereby breaking such users. There's no way to make that experience a good one.
2. Flexibility: if the configuration of individual components isn't exposed to users, we retain the flexibility to change as necessary (e.g. we could switch to nginx if we wanted, but that would be impossible if we made it so that users to modify their apache configs because we can't break our users)
3. Philosophy: the snap is meant to be the way users consume Nextcloud. They shouldn't need to care what web server it uses.

That all said, that's not to say that compression or http2 aren't welcome improvements. It just means that we need to provide that functionality for everyone. What compression config changes would you want to make?

**Comment by kyrofa** [comment]
> when analyzing performance, transporting uncompressed data is the largest time

What tool are you using to determine this, by the way?

**Comment by pachulo** [comment]
@iferca I wanted to try to enable this; what I know is documented here: https://github.com/nextcloud/nextcloud-snap/issues/1297
Let's see if I have some time to try to enable it.

> What tool are you using to determine this, by the way?

I think that he's using [Lighthouse](https://developers.google.com/web/tools/lighthouse/), that comes integrated with Chromium devtools.

**Comment by iferca** [comment]
@pachulo you are right, indeed I used Lighthouse. I was looking for ways to improve responsiveness without increasing my hosting resources. 

![performance](https://user-images.githubusercontent.com/3357362/108270988-e64b1480-714e-11eb-856f-483564072776.png)

> What compression config changes would you want to make?

@kyrofa I was not looking for a particular compression, just looking for what could be improved in terms of performance. Lighthouse reported few things and I thought those were low hanging fruits. I understand your points, although I'm not that familiar with snap, I've found some `nextcloud:occ` and I was hoping I was missing the right ones, so my question here.

> That all said, that's not to say that compression or http2 aren't welcome improvements. It just means that we need to provide that functionality for everyone.

I'd say those should be at least configuration options, wdyt?
